### ISSUE TITLE
port tagger off pkgdb

### ISSUE THREAD

**Issue by pypingou** [fedora-infra/fedora-tagger#200]
Currently tagger uses pkgdb as a source of information for the packages.

Now that pkgdb has been deprecated in Fedora, we should port tagger off pkgdb and to pagure.

**Comment by pypingou** [comment]
This will involve:
- https://github.com/fedora-infra/fedora-tagger/blob/develop/fedoratagger/lib/retired.py (which may need to be ported to PDC actually rather than pkgdb)
- https://github.com/fedora-infra/fedora-tagger/blob/develop/fedoratagger/lib/update.py we could port the yum part to mdapi for simplicity
- https://github.com/fedora-infra/fedora-tagger/blob/develop/fedoratagger/frontend/app.py the app itself links to pkgdb and thus needs updating

**Comment by justwheel** [comment]
A call for maintainers was put out [on the Community Blog](https://communityblog.fedoraproject.org/maintainers-package-tagger/) this morning.

**Comment by illucent** [comment]
Hi! I can try to help with tagger migration as it "is a helpful way to improve search results" application.
A question: April 5, 2018 is deadline for tagger migration or deadline for applying as a contributor / maintainer?
@pypingou I'm okay with Long-term.

**Comment by pypingou** [comment]
@illucent thanks for stepping in, do note that we're looking for someone to take over the long time maintenance of this application, not just migrate it.

**Comment by alick** [comment]
Hi, I'm interested in maintaining tagger (in a long term). I will feel sad to let it go.

Another thing: seems the migration part can be a good idea for GSoC. Are we still in time for that?

**Comment by ekb0** [comment]
Hi. I'll like to help in the migration and maintenance of tagger if anyone would guide me.

**Comment by ctubbsii** [comment]
Wouldn't it be better to just have tags directly on repos in src.fedoraproject.org? GitHub allows tagging repos, pagure could too. Tagger doesn't seem like a necessary separate service now that package repos are a bit more accessible.

**Comment by pypingou** [comment]
@ctubbsii yes and no, pagure already support tagging the projects, but these tags are somewhat different and only people with admin access can tag a project, unlike what tagger offers.

**Comment by pypingou** [comment]
For the people interested in picking up tagger, I invite you to set it up locally and start looking at the different files I pointed out.

If you have any questions, feel free to come #fedora-apps on freenode or starts a thread on the infrastructure mailing list.

**Comment by pypingou** [comment]
Note: since there seems to be a few people interested in taking over tagger, you may want to discuss who works on what to ensure there is as little overlap as possible between what you all work on :)

**Comment by illucent** [comment]
I can focus on front-end (Web UI), not installed locally yet.

**Comment by pypingou** [comment]
Is there any progress on this?
Anything we could do to help?

**Comment by illucent** [comment]
I do not know, currently any progress, I have thought there can be some discussion at irc channel, waiting for others.

**Comment by pypingou** [comment]
If someone organizes a meeting that would be doable :)

**Comment by alick** [comment]
I wanna focus on the maintaining side, i.e. running the sevice on infra.

**Comment by ekb0** [comment]
When should we hold a meeting? I will be available for a meeting at the channel on Friday, 8:00 PM UTC +0 but my time is adjustable

**Comment by justwheel** [comment]
@illucent @alick @elormes Does someone want to take lead on organizing a meeting and establishing an agenda?

If the three of you are interested in working with the application, I would start by finding a time to hold a quick IRC meeting and put it on the Fedora calendar. To find a mutual time, one of you can create a [WhenIsGood survey](http://whenisgood.net/) and send that to interested folks.

Once you know a good day and time, add the meeting date and time to the [Fedora calendar](https://apps.fedoraproject.org/calendar/). You can reserve an IRC meeting channel like `#fedora-meeting` or `#fedora-meeting-1` this way.

In your meeting, you can identify key areas of focus to get the Tagger off of "life support" and how you want to divide the work. If you have questions, I think folks from Infrastructure will be happy to answer and guide you, but the team doesn't have time to lead the organization for future development.

I don't have a ton of time to help much here, but I'm happy to help out and get things rolling if you need help. Otherwise, it would be great if someone can take lead here and work on setting up a time to take the project forward.

**Comment by ekb0** [comment]
Click [here](http://whenisgood.net/mzzj2zk) to access the WhenIsGood survey. Please fill in the times you will be available.

**Comment by alick** [comment]
What is the timezone in use for the survey?


Regards,

**Comment by ekb0** [comment]
Thanks for asking. I thought I had ticked use timezones. I have now enabled it so please take the survey again and select your location.

**Comment by alick** [comment]
Do we have a result of the survey?


Regards,

**Comment by ekb0** [comment]
So far, we have Friday, 2:00 PM (UTC)

**Comment by alick** [comment]
Good. Talk to you then. Let me know which IRC channel we're gonna use.

**Comment by pypingou** [comment]
Let's ensure this is properly announced

**Comment by justwheel** [comment]
@elormes @alick +1, if the meeting time among interested maintainers is agreed, definitely get it onto the [Fedora calendar](https://apps.fedoraproject.org/calendar/). I would also announce the meeting time and IRC channel on the [infrastructure mailing list](https://lists.fedoraproject.org/admin/lists/infrastructure.lists.fedoraproject.org/) too.

**Comment by ekb0** [comment]
I'm not allowed to add new meetings to the Fedora infrastructure calendar

**Comment by justwheel** [comment]
@elormes Oops, I think this is because you may not have CLA+1 yet. I added it for you:

https://apps.fedoraproject.org/calendar/meeting/9217/

**Comment by ekb0** [comment]
Thanks. I've announced it in the mailing list too

**Comment by illucent** [comment]
missed today's meeting

**Comment by cverna** [comment]
@illucent you can check the minutes and full logs here --> https://meetbot.fedoraproject.org/fedora-meeting-1/2018-04-20/fedora_tagger_-_future_development.2018-04-20-14.27.html

**Comment by illucent** [comment]
@cverna Thanks!
